### DISCUSSION TITLE
change ip method header index in request object

### DISCUSSION THREAD

**Discussion by AliGhaleyan** [laravel/framework#35630] [General] [answered]
i want to change the default index for get client ip

when run:
```php
<?php
$request->ip();
```
default index for get ip from headers is `REMOTE_ADDR`
i want to be another index like `X_FORWARDED_FOR`  or ....

**Comment by kohlerdominik** [comment, marked as answer]
you should use TrustProxies middleware and add your trusted proxies there. 
https://laravel.com/docs/8.x/requests#configuring-trusted-proxies

Then, `$request->ip()` will return the correct ip.

**Reply by AliGhaleyan** [reply]
tnx it worked 👍
